### ISSUE TITLE
Step Through the Simulation and Display the Results

### ISSUE THREAD

**Issue by tlooy** [jtigger/kanban-simulator#3]
# As the Kanban Simulation Administrator I want to be able to step through the simulation one cycle at at time and see the results of each step by having the simulator display the story cards that are in each workflow queue.
# Assumptions

A step is the action the administrator takes to cycle the workflow system once.  A cycle is the advancing of story cards through the system one story point at a time.  A Story Card will advance to the next queue based on the capacity of WIP Limits on each worflow's queue and the Story Point Estimate on the Story Card.  In other words, a Story Card does not advance to the next queue until it has been in its current queue for the number of cycles that corresponds to its Story Point Estimate and the next queue has capacity for another Story Card.

It is assumed that the Story Card Estimate value is applicable for each workflow queue.  Later we will play a story that allows us to adjust the level of effort for a Story Card for each workflow queue.

When the simulation starts, the first workflow queue (In Analysis) will take enough story cards out of the backlog to fill up its WIP Limit.   
# Details

A Story Card cannot advance to the next queue until it has been in its current queue for the number of cycles that corresponds to its Story Point Estimate.

A Story Card cannot advance to the next queue unless there is capacity in the queue for another Story Card.  (The WIP Limit is what determines the capacity of the workflow queue.)

A Story Card cannot skip workflow queues, the must go through each queue and in order.

When a Story Card advances into a new Workflow Queue the Story Card's cycle count for that queue is set to zero.  Each subsequent cycle will increment it's queue cycle count by one.   Only after the queue cycle count for a Story Card is equal to or greater than it's Story Point Estimate is it eligible to be promoted to the next workflow queue. 

The workflow queues will be worked from right to left in a pull fashion.  In other words, the Done queue will go first, looking to see if there are any Story Cards in the In Test queue that are ready to be pulled into Done.  Next, the In Test queue will pull as many eligible Story Cards from Read For Test that it has capacity for.  And so on...
# Acceptance Criteria

**Comment by jtigger** [comment]
Tom and I discussed how we'd like the output of the simulator to be UI-agnostic and suitable to fuel both a text-based output, a GUI, and a replay mechanism (e.g. animate a PowerPoint presentation or a Flash).

From an implementation perspective, I envision a transaction log type output.  Details will naturally fall-out during coding.
